activerecord/lib/active_record/railtie.rb

```diff
@@ -31,6 +32,8 @@ class Railtie < Rails::Railtie # :nodoc:
     config.active_record.sqlite3 = ActiveSupport::OrderedOptions.new
     config.active_record.sqlite3.represent_boolean_as_integer = nil
 
+    config.active_record.queues = ActiveSupport::InheritableOptions.new(destroy: :active_record_destroy)
```

Comment:
More than a few. ;)

We like sending built-in jobs to separate queues by default because it allows us to prioritize them. For example, we can configure a pool of Resque workers to serve jobs in the following prioritized order:

1. `chat` (In-app jobs that write new chat lines to the DB; most sensitive to delay)
1. `action_mailbox_routing`
2. `active_storage_analysis`
3. `search` (In-app jobs that update Elasticsearch indexes)
4. `active_record_destroy`
5. `active_storage_purge` (Least sensitive to delay)

Using separate queues also allows us to stop work on a particular queue when the jobs it hosts put too much pressure on the DB or an external service.

The guides should probably list the default queues for built-in jobs and explicitly state that it may be necessary to configure the Active Job backend to serve them. I also want to update the API docs for each of the built-in jobs to mention their respective default queues.

Reply:
All good reasons, thanks! Agreed with the guides, might be nice to add the advice about serving queues in prioritized order as well.